Always provide encoding in MetaData constructor

oscargus opened the pull request:
Unfortunately there was still a `Globals` dependency recently introduced...

tobiasdiez commented on src/main/java/net/sf/jabref/MetaData.java:
```diff
@@ -83,26 +83,17 @@
 
     /**
-     * The MetaData object can be constructed with no data in it.
+     * The MetaData object can be constructed with no data in it, but needs encoding information
```
Why is the encoding special and always has to be provided while other metadata is optional? I would still provide an empty constructor and let `getEncoding` return an `Optional<Encoding>`. Then it is the caller business to decide which encoding he wants if no encoding is set in the metadata.

oscargus commented:
Clearly a better idea. :+1: There are quite often cases where we need to provide "unnecessary" data unfortunately, but then that's no reason to increase them.

oscargus commented:
OK, I changed that now. I also changed `ImportFormat` to always take an `ImportFormatPreferences` as an argument to the constructor. This was really required to be able to test `CustomImports` and I think it makes sense for custom import format developers to have some access to preferences. Not sure if I should provide an argument-less constructor as well, creating an `ImportFormatPreferences` from `Globals`, just to ease the transition (also, people may have old custom import formats that they do not have source for and so on...). WDYT?

oscargus commented:
I'll try to redo this in smaller steps.
